dashboard/apps/prototype/views.py

```diff
@@ -41,8 +47,8 @@ def project_json(request):
     try:
         project = Project.objects.visible().get(
             id=request_data['projectid'])
-    except Project.DoesNotExist:
-        return HttpResponseNotFound(
-            'cannot find project with projectid={}'
-            .format(request_data['projectid']))
+    except (ValueError, Project.DoesNotExist):
+        error = 'cannot find project with projectid={}'.format(
+            request_data['projectid'])
+        return JsonResponse({'error': error}, status=404)
```

Comment:
Do we need to handle the possibility of people accessing projects which should not be visible through manually entering the project_id in the url?